client/state/selectors/is-mapped-domain-site.js

```diff
@@ -2,27 +2,32 @@
  * External dependencies
  */
 
-import { get } from 'lodash';
+import { get, some } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import getRawSite from 'state/selectors/get-raw-site';
+import { getDomainsBySiteId } from 'state/sites/domains/selectors';
 
 /**
  * Returns true if site is a mapped domain site, false if the site is not,
  * or null if the site is unknown.
  *
- * @param {Object} state Global state tree
- * @param {Number} siteId Site ID
- * @return {?Boolean} Whether site is a mapped domain site
+ * @param {object} state Global state tree
+ * @param {number} siteId Site ID
+ * @returns {?boolean} Whether site is a mapped domain site
  */
 export default function isMappedDomainSite( state, siteId ) {
 	const site = getRawSite( state, siteId );
+	const domains = getDomainsBySiteId( state, siteId );
 
-	if ( ! site ) {
+	if ( ! site || 0 === domains.length ) {
 		return null;
 	}
 
-	return get( site, 'options.is_mapped_domain', false );
+	return (
+		get( site, 'options.is_mapped_domain', false ) &&
+		some( domains, ( { isWPCOMDomain } ) => ! isWPCOMDomain )
```

Comment:
Looks like this additional check pulled in some cases for Jetpack sites. It might be worth double-checking other places this selector is used or use a different approach.